### PR TITLE
Algolia Field Groups to Support Event Forwarding and Source Connector

### PR DESCRIPTION
**Pull request by smomin**
- Algolia Event supports the Algolia Event Forward Extension to pull user events from the site and store it in AEP for use
- Algolia Profile supports the Algolia User Profile Source connector to pull profile to be stored in the realtime CDP to build audiences

### CODE REVIEW

File: extensions/algolia/fieldgroups/algolia-event.schema.json
Comment (anandphatak) [inline extensions/algolia/fieldgroups/algolia-event.schema.json:17]:
```suggestion
    "algolia-event": {
          "type": "object",
          "properties": {
            "algoliaEvent": {
```
We need to add a namespace for algolia event fields so that they won't collide with other experience event fields. To add a namespace, please wrap the entire schema definition inside an object "algoliaEvent"

File: extensions/algolia/fieldgroups/algolia-event.schema.json
Comment (anandphatak) [inline extensions/algolia/fieldgroups/algolia-event.schema.json:18]: You would need to move the "required" array inside the algoliaEvent object after making the suggested changes above.

File: extensions/algolia/fieldgroups/algolia-event.example.1.json
Comment (anandphatak) [inline extensions/algolia/fieldgroups/algolia-event.example.1.json:2]:
```suggestion
  "algoliaEvent": {
```
After making the changes suggested for `algolia-event.schema.json`, the example will change as specified above.

File: extensions/algolia/fieldgroups/algolia-profile.schema.json
Comment (anandphatak) [inline extensions/algolia/fieldgroups/algolia-profile.schema.json:17]:
```suggestion
    "algolia-profile": {
        "type": "object",
        "properties": {
          "algoliaProfile": {
          "type": "object",
```
We need to add a namespace for algolia profile fields so that they won't collide with other experience event fields. To add a namespace, please wrap the entire schema definition inside an object "algoliaProfile"

File: extensions/algolia/fieldgroups/algolia-profile.example.1.json
Comment (anandphatak) [inline extensions/algolia/fieldgroups/algolia-profile.example.1.json:2]:
```suggestion
  "algoliaProfile": {
```
After making the changes suggested for algolia-profile.schema.json, the example will change as specified above.

File: extensions/algolia/fieldgroups/algolia-profile.schema.json
Comment (smomin) [reply extensions/algolia/fieldgroups/algolia-profile.schema.json:17]: I have updated the schema.

File: extensions/algolia/fieldgroups/algolia-profile.example.1.json
Comment (smomin) [reply extensions/algolia/fieldgroups/algolia-profile.example.1.json]: I have updated the schemas with the suggested namespace.

File: extensions/algolia/fieldgroups/algolia-event.schema.json
Comment (smomin) [reply extensions/algolia/fieldgroups/algolia-event.schema.json]: I have moved the required properties into the correct nested object.

**Comment by smomin** [comment]
@anandphatak do you have anymore feedback?

**Comment by smomin** [comment]
@anandphatak Checking on this PR.  any more comments?

**Comment by anandphatak** [comment]
Hi @smomin , I think the PR looks good. We can merge it; but release to production will happen only in January '25

**Comment by smomin** [comment]
hey @anandphatak Jan 25 is fine.  do you execute the merge or do I wait for approval so I can complete the merge?

**Comment by anandphatak** [comment]
> hey @anandphatak Jan 25 is fine. do you execute the merge or do I wait for approval so I can complete the merge?

I will do the merge.

**Review by anandphatak — APPROVED** [review]

**Pull request merged**
